TCL4 Data Management/utm-tcl4-dmp-delete-api.yaml

```diff
@@ -0,0 +1,180 @@
+    type: oauth2
+    tokenUrl: 'https://utm.defined.host/oauth/token'
+    flow: application
+    scopes:
+      'utm.nasa.gov_write.message': Subject can read, create, and update message data such as UTM Message and Negotiations and DMP data
+
+paths:
+  '/delete-by-callsign-run-model':
+    delete:
+      summary: Delete ALL data for a particular (callsign, run, model) triple.
+      description: |-
+        This endpoint should be used with care. If the USS owns the data for
+        a particular callsign-run pairing, then it can request to delete all
+        previously submitted data for a given model. This is usually needed
+        if a USS catches a problem with the data that it submitted and needs
+        to fix the error in its data and resubmit.
+
+        Due to the way in which data are processed on the backend, a USS MUST
+        wait an hour after a successful call to this endpoint before trying
+        to submit the replacement data.  This is due to the potential for the
+        replacement data to get persisted prior to the DELETE being executed.
+
+        Note that this deletes ALL records for a given model for that callsign
+        and run.  So if only a particular instance was 'bad' the USS would need
+        to re-send all instances with the fixed instance in order to ensure
+        completeness of data.
+
+        Ideally, the USS will notify NASA of the need to use this endpoint
+        prior to calling it so that we can monitor the data processes on the
+        backend.
+      security:
+        - ussapi_security:
+            - utm.nasa.gov_write.message
+      parameters:
+        - in: body
+          name: DeleteDmpData
+          required: true
+          schema:
+            $ref: '#/definitions/CallsignRunModel'
+
+      responses:
+        '202':
+          description: Accepted. Assume success, but verify with NASA.
+          schema:
+            $ref: >-
+              https://raw.githubusercontent.com/nasa/utm-apis/v4-draft/utm-domains/utm-domain-commons.yaml#/definitions/UTMRestResponse
+        '401':
+          description: Invalid or missing access_token provided.
+          schema:
+            $ref: >-
+              https://raw.githubusercontent.com/nasa/utm-apis/v4-draft/utm-domains/utm-domain-commons.yaml#/definitions/UTMRestResponse
+        '403':
+          description: Forbidden. Do not retry with same access token. Reason not provided, but do you have the right scopes? Does uss_name match the sub claim?
+          schema:
+            $ref: >-
+              https://raw.githubusercontent.com/nasa/utm-apis/v4-draft/utm-domains/utm-domain-commons.yaml#/definitions/UTMRestResponse
+definitions:
+  CallsignRunModel:
+    description: >-
+      The callsign, run, and model to be be deleted.
+    type: object
+    required:
+      - call_sign
+      - test_run
+      - model
+    example:
+      call_sign: Viper1
+      test_run: 2045
+      model: FlightEssentialTelemetryAndUasStates
```

Comment:
I think the api end point here would be better - it would be the same endpoint that was used to upload data associated with the (call_sign, test_run) pair.  That way we can use the same mappings from before to UDS

Reply:
i don't understand your comment.  where is the "api end point" going to be better?

Reply:
the protocol was based on endpoint not data model - however this is no longer relevant since we are not pursuing this endpoint